### PR TITLE
Add the option to specify preboot configurations

### PR DESCRIPTION
**Pull request by julienduchesne**
These configurations are either variable files or terragrunt configs

### CODE REVIEW

**Comment by julienduchesne** [comment]
@jocgir It's actually ready now (not WIP anymore). I added support for S3 paths and some debug logs (config files as they are read before and after gotemplate)

File: test/fixture-bootstrap/chain-preboot-configs/variables.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+    "my_map": {
+        "key1": "value1",
+        "key2": "value2"
+    }
+}
```
Comment (jocgir) [inline test/fixture-bootstrap/chain-preboot-configs/variables.json]:
```suggestion
}

```

File: test/fixture-bootstrap/preboot/variables.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+    "my_variable": "my_value",
+    "my_variable2": "my_value2"
+}
```
Comment (jocgir) [inline test/fixture-bootstrap/preboot/variables.json]:
```suggestion
}

```

File: test/fixture-bootstrap/refer-to-other-source/terragrunt_files/.tgf.config [hunk 3]
```diff
@@ -0,0 +1 @@
+docker-image-version: 2.8
```
Comment (jocgir) [inline test/fixture-bootstrap/refer-to-other-source/terragrunt_files/.tgf.config]:
```suggestion
docker-image-version: 2.8

```

Comment (julienduchesne) [reply test/fixture-bootstrap/refer-to-other-source/terragrunt_files/.tgf.config]: Oops, I actually forgot the .tgf.config. It shouldn't be there

**Pull request merged**